src/flake8_simplify/rules/ast_with.rs

```diff
@@ -1,22 +1,56 @@
-use rustpython_ast::{Stmt, StmtKind};
+use rustpython_ast::{Stmt, StmtKind, Withitem};
 
+use crate::ast::helpers::first_colon_range;
 use crate::ast::types::Range;
 use crate::checkers::ast::Checker;
 use crate::registry::Diagnostic;
 use crate::violations;
 
-/// SIM117
-pub fn multiple_with_statements(checker: &mut Checker, stmt: &Stmt) {
-    let StmtKind::With { body, .. } = &stmt.node else {
-        return;
-    };
+fn find_nested_with(body: &[Stmt]) -> Option<(&Vec<Withitem>, &Vec<Stmt>)> {
     if body.len() != 1 {
-        return;
+        return None;
+    }
+    let StmtKind::With { items, body, .. } = &body[0].node else {
+        return None
+    };
+    find_nested_with(body).or(Some((items, body)))
+}
+
+/// SIM117
+pub fn multiple_with_statements(
+    checker: &mut Checker,
+    stmt: &Stmt,
+    body: &[Stmt],
+    parent: Option<&Stmt>,
+) {
```

Comment:
I think this signature has gotten complex enough to warrant a doc comment that explains what these various `Stmt`s should be.

Reply:
can we rename variable names?

Reply:
Renamed the arguments.